tchannel/tornado/message_factory.py

```diff
@@ -239,94 +236,120 @@ def build_response(self, message):
         )
         return res
 
-    def build_context(self, message):
-        if message.message_type == Types.CALL_REQ:
-            return self.build_request(message)
-        elif message.message_type == Types.CALL_RES:
-            return self.build_response(message)
+    def build_inbound_response(self, message, response):
```

Comment:
So if I'm reading this correctly, the state management here is happening like this:

    message = # received new call_res
    response = factory.build_inbound_response(message, None)

    while fragmentation bit is set:
        message = # received new call_res_continue
        factory.build_inbound_response(message, response)

This is inconsistent. `build_inbound_response`'s return value is only used once. This feels like a bad way of managing this state. It would be more consistent if `build_inbound_response` always returned a response but that's pointless if it's mutating the arg stream anyway.

If you have to plumb state like this, maybe message factory should be managing it after all?